test/jpypetest/test_fault.py

```diff
@@ -0,0 +1,1249 @@
+        with self.assertRaisesRegex(SystemError, "fault"):
+            jo.substring
+        jo.substring
+
+    @common.requireInstrumentation
+    def testJPObject_setattro(self):
+        jo = JString("f")
+        _jpype.fault("PyJPObject_setattro")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            jo.substring = None
+
+    @common.requireInstrumentation
+    def testJPField(self):
+        jf = JClass("jpype.common.Fixture")
+        jfi = jf()
+        with self.assertRaises(AttributeError):
+            jf.final_static_int_field = 2
+        with self.assertRaises(AttributeError):
+            jfi.final_int_field = 2
+        _jpype.fault("JPField::setStaticAttribute")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            jf.static_int_field = 2
+        _jpype.fault("JPField::setAttribute")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            jfi.int_field = 2
+        _jpype.fault("JPField::getStaticAttribute")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            i = jf.static_int_field
+        _jpype.fault("JPField::getAttribute")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            i = jfi.int_field
+        si = jf.__dict__['static_int_field']
+        str(si)
+        repr(si)
+        i = None
+        _jpype.fault("PyJPField_get")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            i = jfi.int_field
+        self.assertEqual(i, None)
+        _jpype.fault("PyJPField_set")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            jfi.int_field = 2
+        _jpype.fault("PyJPField_repr")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            repr(si)
+
+    @common.requireInstrumentation
+    def testConvertString(self):
+        _jpype.fault("JPObjectType::canConvertToJava")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JObject._convertToJava("foo")
+        _jpype.fault("JPConversionString::matches")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JString._convertToJava("foo")
+
+    @common.requireInstrumentation
+    def testJPObject(self):
+        jf = JClass("jpype.common.Fixture")
+        jfi = JClass("jpype.common.Fixture")()
+        _jpype.fault("JPClass::setStaticField")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            jf.static_object_field = None
+        _jpype.fault("JPClass::setField")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            jfi.object_field = None
+        i = None
+        _jpype.fault("JPClass::getStaticField")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            i = jf.static_object_field
+        _jpype.fault("JPClass::getField")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            i = jfi.object_field
+        self.assertEqual(i, None)
+
+    @common.unittest.SkipTest
```

Comment:
Skip for which reason? Please add a FIXME or TODO

Reply:
Will do.   Most of the skip tests are because something different happened in the CI than happened on my system.  In particular those for the jvmfinder are all working tests on my system that fail or hang the CI.  (I believe it is a problem with mock versions.)  

I am hoping when I finish this pass the ^C crash issue on Spyder will fix.

Reply:
In my experience the mock library is very stable. Can you please point me to a particular test, which fails only on CI, but succeeds on your system?

Reply:
That was stupid, you have decorated the one already :)